content/en/blog/_posts/2021-12-15-pod-security-admission-beta.md

```diff
@@ -0,0 +1,778 @@
+kind-control-plane   Ready    control-plane,master   54m   v1.23.0
+```
+
+### Confirm Pod Security is enabled
+
+The best way to [confirm the API's default enabled plugins](/docs/reference/access-authn-authz/admission-controllers/#which-plugins-are-enabled-by-default) is to check the Kubernetes API container's help arguments.
+
+```shell
+kubectl -n kube-system exec kube-apiserver-kind-control-plane -it -- kube-apiserver -h | grep "default enabled ones"
+```
+
+The output is similar to this:
+
+```
+...
+      --enable-admission-plugins strings
+admission plugins that should be enabled in addition
+to default enabled ones (NamespaceLifecycle, LimitRanger,
+ServiceAccount, TaintNodesByCondition, PodSecurity, Priority,
+DefaultTolerationSeconds, DefaultStorageClass,
+StorageObjectInUseProtection, PersistentVolumeClaimResize,
+RuntimeClass, CertificateApproval, CertificateSigning,
+CertificateSubjectRestriction, DefaultIngressClass,
+MutatingAdmissionWebhook, ValidatingAdmissionWebhook,
+ResourceQuota).
+...
+```
+
+`PodSecurity` is listed in the group of default enabled admission plugins.
+
+If using a cloud provider, or if you don't have access to the API server, the best way to check would be to run a quick end-to-end test:
+
+```shell
+kubectl create namespace verify-pod-security
+kubectl label namespace verify-pod-security pod-security.kubernetes.io/enforce=restricted
+# The following command does NOT create a workload (--dry-run=server)
+kubectl -n verify-pod-security run test --dry-run=server --image=busybox --privileged
+kubectl delete namespace verify-pod-security
+```
+
+The output is similar to this:
+
+```
+Error from server (Forbidden): pods "test" is forbidden: violates PodSecurity "restricted:latest": privileged (container "test" must not set securityContext.privileged=true), allowPrivilegeEscalation != false (container "test" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "test" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or container "test" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "test" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
+```
+
+### Configure Pod Security
+
+Policies are applied to a namespace via labels. These labels are as follows:
+   * `pod-security.kubernetes.io/<MODE>: <LEVEL>` (required to enable pod security)
+   * `pod-security.kubernetes.io/<MODE>-version: <VERSION>` (*optional*, defaults to latest)
+
+A specific version can be supplied for each enforcement mode. The version pins the policy to the version that was shipped as part of the Kubernetes release. Pinning to a specific Kubernetes version allows for deterministic policy behavior while allowing flexibility for future updates to Pod Security Standards. The possible <MODE(S)> are `enforce`, `audit` and `warn`.
+
+### When to use `warn`?
+
+The typical uses for `warn` are:
```

Comment:
```suggestion
The typical uses for `warn` are to get ready for a future change where you want to enforce
a different policy. The most two common cases would be:
```
How about case 3, you're not enforcing any policy yet, and you want to preview the impact? I think that's a good use for `warn` (maybe also: delete all existing Pods once you do).

I should add that I haven't actually tried Pod Security admission in any real-world cluster with a real world workload on it, so please take that in mind when considering my suggestions.

Reply:
> How about case 3, you're not enforcing any policy yet, and you want to preview the impact? I think that's a good use for warn

That's sort of the same as "`warn` at a stricter level", where `enforce` is privileged and `warn` is baseline or restricted. If you think it's clearer to explicitly call that out as a third case, that's fine

> (maybe also: delete all existing Pods once you do).

If you're looking to check a new policy non-disruptively, deleting existing pods is not the way to do it :-)